src/include/platform/internal/GenericPlatformManagerImpl_CMSISOS.ipp

```diff
@@ -0,0 +1,400 @@
+                {
+                    ChipLogError(DeviceLayer, "Error handling CHIP timers: %" CHIP_ERROR_FORMAT, err.Format());
+                }
+            }
+        }
+        else
+        {
+            // No CHIP timers are active, so wait indefinitely for an event to arrive on the event
+            // queue.
+            waitTimeInTicks = osWaitForever;
+        }
+
+        // Unlock the CHIP stack, allowing other threads to enter CHIP while
+        // the event loop thread is sleeping.
+        StackUnlock unlock;
+        osStatus_t eventReceived = osMessageQueueGet(mChipEventQueue, &event, nullptr, waitTimeInTicks);
+
+        // If an event was received, dispatch it and continue until the queue is empty.
+        while (eventReceived == osOK)
+        {
+            StackLock lock;
+            Impl()->DispatchEvent(&event);
+            StackUnlock unlock;
+            eventReceived = osMessageQueueGet(mChipEventQueue, &event, nullptr, 0);
+        }
+    }
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericPlatformManagerImpl_CMSISOS<ImplClass>::_StartEventLoopTask(void)
+{
+    mEventLoopTask = osThreadNew(EventLoopTaskMain, this, &mEventLoopTaskAttr);
+    return (mEventLoopTask != nullptr) ? CHIP_NO_ERROR : CHIP_ERROR_NO_MEMORY;
+}
+
+template <class ImplClass>
+void GenericPlatformManagerImpl_CMSISOS<ImplClass>::EventLoopTaskMain(void * pvParameter)
+{
+    ChipLogDetail(DeviceLayer, "CHIP event task running");
+    static_cast<GenericPlatformManagerImpl_CMSISOS<ImplClass> *>(pvParameter)->Impl()->RunEventLoop();
+}
+
+/**
+ * @brief Calculate the elapsed time of the active chip platform timer since it has been started,
+ *        as set in mNextTimerBaseTime, and adjust its remaining time in mNextTimerDurationTicks member
+ *
+ * @return The next Timer remaining time in ticks
+ */
+template <class ImplClass>
+uint32_t GenericPlatformManagerImpl_CMSISOS<ImplClass>::SyncNextChipTimerHandling()
+{
+    uint32_t elapsedTime;
+    uint32_t timerBaseTime = mNextTimerBaseTime;
+    uint32_t currentTime   = osKernelGetTickCount();
+    if (currentTime < timerBaseTime)
+    {
+        // TickCount has wrapped around
+        elapsedTime = (UINT32_MAX - timerBaseTime) + currentTime;
+    }
+    else
+    {
+        elapsedTime = currentTime - timerBaseTime;
+    }
+
+    if (elapsedTime < mNextTimerDurationTicks)
+    {
+        // We didn't timeout yet, adjust the remaining time
+        mNextTimerDurationTicks -= elapsedTime;
+        mNextTimerBaseTime = osKernelGetTickCount();
+    }
+    else
+    {
+        mNextTimerDurationTicks = 0;
+    }
+    return mNextTimerDurationTicks;
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericPlatformManagerImpl_CMSISOS<ImplClass>::_PostBackgroundEvent(const ChipDeviceEvent * event)
+{
+#if defined(CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING) && CHIP_DEVICE_CONFIG_ENABLE_BG_EVENT_PROCESSING
+    if (mBackgroundEventQueue == NULL)
+    {
+        return CHIP_ERROR_INTERNAL;
+    }
```

Comment:
```suggestion
VerifyOrReturnError(mBackgroundEventQueue != nullptr, CHIP_ERROR_INTERNAL);
```

Should this also be the not-ini error code?

Reply:
It could, It was returning this error code in the freeRTOS version so I didn't change it